viewing_party/party.py

```diff
@@ -0,0 +1,243 @@
+    #convert difference set to tuple to iterate and make new dictionary
+    movie_titles_unqiue = tuple(my_movie_set - friend_movie_set)
+
+    #loop through each title in movies_title_unquie tuple and add title
+    #dictionary to unser_unique_movie_watched list
+    for title in movie_titles_unqiue:
+        user_unqiue_movie_watched.append({'title': title})
+
+    return user_unqiue_movie_watched
+
+#take movies from user and return set of movie titles from user list
+def make_my_movie_set(user_data):
+    #intialize variable
+    user_movie_set = set()
+
+    #for each movie in watched list, add only movie title to user_set set
+    for my_movies in user_data['watched']:
+        user_movie_set.add(my_movies['title'])
+    
+    return user_movie_set
+
+#takes nested friend dictionary list and returns set of movie titles from friend list
+def make_friend_movie_set(user_data):
+    #intialize variables
+    friend_movie_set = set()
+    friend_list = user_data['friends']
+
+    #iterating thru each friend's movie dictionary list
+    #set friend_movie_list to hold the dictionary of of movies watched from friends
+    for friend_num in range(len(friend_list)):
+        friend_movie_list = friend_list[friend_num]['watched']
+        #for each movie in friend_movie_list, add movie title to friend_movie_set
+        for movie in friend_movie_list:
+            friend_movie_set.add(movie['title'])
+
+    return friend_movie_set
+
+#returns list of dictionary that friends have seen that user has not seen
+def get_friends_unique_watched(user_data):
+    #initalize variables
+    friend_unqiue_movie_watched = []
+
+    #call set-making functions and capture return in assigned var
+    my_movie_set = make_my_movie_set(user_data) 
+    friend_movie_set = make_friend_movie_set(user_data)
+    #convert difference set to tuple to iterate and make new dictionary
+    movie_titles_unqiue = tuple(friend_movie_set - my_movie_set)
+    
+    #add title to unique_movie_watched list
+    for title in movie_titles_unqiue:
+        friend_unqiue_movie_watched.append({'title': title})
+
+    return friend_unqiue_movie_watched
+
+######################################################
+                # Wave 4 #
+######################################################
+
+#takes user_data and returns a list of reccomended movies referencing
+#friend's movie watched list and user subscription
+def get_available_recs(user_data):
+    #intialize variables
+    rec_list = []
+    movie_title_user = []
+
+    #add movie title to movie_title_user list
+    for movie_user in user_data['watched']:
+        movie_title_user.append(movie_user['title'])
+
+    #access friend key in dict for friend list
+    for friend in user_data['friends']:
+        friend_movies_watched_list = friend['watched']
+
+        #make boolean list from friend watched list
+        for movie_item in friend_movies_watched_list:
+            user_has_seen_movie = movie_item['title'] in movie_title_user
```

Comment:
Nice use of temp variables with boolean values. This is a nice way to break down the problem